pkg/cloudprovider/aws/fleet/capacity.go

```diff
@@ -99,3 +100,22 @@ func (c *Capacity) GetTopologyDomains(ctx context.Context, key cloudprovider.Top
 		return nil, fmt.Errorf("unrecognized topology key %s", key)
 	}
 }
+
+func (c *Capacity) Delete(ctx context.Context, nodes []*v1.Node) ([]*string, error) {
+	ids := c.instanceProvider.getInstanceIDs(ctx, nodes)
+
+	output, err := c.instanceProvider.ec2.TerminateInstancesWithContext(ctx, &ec2.TerminateInstancesInput{
+		DryRun:      ptr.Bool(false),
+		InstanceIds: ids,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("terminating ec2 instance, %w", err)
```

Comment:
I can imagine it would change based on the cloud provider, but the EC2 instances here all get deleted at the same time. So this will only return an error after all of the EC2 delete commands have been initiated.